### DISCUSSION TITLE
Custom control with reference to form components

### DISCUSSION THREAD

**Discussion by mravell** [formio/formio.js#4383] [Q&A] [answered]
Is it possible, when writing a custom component, that I can get a reference to the form that is being built within the edit screen of the component?

i.e. I want a form component to be able to have a list of the other components that are already in the form that is being built.

Does that make sense? The context object does not seem to have any reference to the form.

{
	type: 'select',
	input: true,
	key: 'testComp',
	label: 'Test',
	tooltip: 'Select the component to source the value from on submit.',
	dataSrc: 'custom',
	weight: 10,
	data: {
		custom: function custom(context) {
			var values = [];
			// I want to get a list of the other components in the form here and add them into the values array
			// context object doesn't seem to have what we need as it is the context of the edit form.

			return values;
		}
	}
},

**Comment by mravell** [comment, marked as answer]
This can be closed.

A reference to the other form fields is available in builder.instance.schema.components.

**Reply by muneebahmad0600** [reply]
Need an example for this if possible.

here is a snippet from developer tools. I don't see "builder" in the "context" variable.
![image](https://github.com/formio/formio.js/assets/38345528/78ff685b-6568-47be-83c0-68ea7108ba8c)

Not able to find any documentation. may be I need to add an input? 
Any help would be very much appreciated.. Thanks in advance.

**Reply by dinbtechit** [reply]
Hi @mravell, Just curious.. how did you access the `builder` object? 

```
{
 type: 'select',
 input: true, 
 key: 'testComp',
 label: 'Test',
 tooltip: 'Select the component to source the value from on submit.',
 dataSrc: 'custom',
 weight: 10,
 data: {
   custom: function custom(context) {
    // builder is undefined here.
 }
},
```

**Comment by bilogic** [comment]
@mravell can you give an example please?
